### DISCUSSION TITLE
[Help]: Manually Trigger Embla Carousel Options

### DISCUSSION THREAD

**Discussion by dpaulos6** [davidjerleke/embla-carousel#768] [Help]
### Summary

I'm implementing a radial progress bar for each slide, like slowly filling the circle until it auto slides, and I need the timer to reset every time I click the next/prev buttons and auto scroll when this timer ends.
Everything related to the timer works as it should, but I need help to trigger the scroll next when I need it and do something when one of the buttons is clicked.

I've tried using `emblaApi?.scrollNext()` but it doesn't do anything, and on the docs, I couldn't find any information about the next/prev buttons.

The code is on **Aditional Information**

PS: This is a bit urgent for me because it is part of my internship and I would appreciate an answer asap.

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

```tsx
import React, { useCallback, useEffect, useState } from "react";
import { EmblaOptionsType, EmblaCarouselType, EmblaEventType } from "embla-carousel";
import useEmblaCarousel from "embla-carousel-react";
import ClassNames from "embla-carousel-class-names";
import { usePrevNextButtons } from "./EmblaCarouselArrowButtons";
import { ChevronLeft, ChevronRight } from "lucide-react";
import { Button } from "@/components/ui/button";
import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import Autoplay from "embla-carousel-autoplay";

type SlideType = {
  src: string;
  alt: string;
};

type PropType = {
  slides: SlideType[];
  options?: EmblaOptionsType;
  time: number
};


const EmblaCarousel: React.FC<PropType> = (props) => {
  const { slides, options, time } = props;
  const timeMilli = time * 1000;
  const [emblaRef, emblaApi] = useEmblaCarousel(options, [ClassNames()]);
  const [percentage, setPercentage] = useState<number>(0);

  const {
    prevBtnDisabled,
    nextBtnDisabled,
    onPrevButtonClick,
    onNextButtonClick,
  } = usePrevNextButtons(emblaApi);
  
  useEffect(() => {
    let timer: NodeJS.Timeout;
    let startTime: number = Date.now();
    
    const updatePercentage = () => {
      const currentTime = Date.now();
      const elapsedTime = currentTime - startTime;
      const remainingTime = Math.max(timeMilli - elapsedTime, 0);
      let newPercentage = 0;
      
      if (remainingTime > 0) {
        newPercentage = ((timeMilli - remainingTime) / timeMilli) * 100;
      } else {
        startTime = Date.now();
        emblaApi?.scrollNext();
        timer = setTimeout(updatePercentage, 100);
      }
  
      setPercentage(newPercentage);
      
      if (newPercentage < 100) {
        timer = setTimeout(updatePercentage, 100);
      }
    };
    
    updatePercentage();
    
    return () => clearTimeout(timer);
  }, [timeMilli]);
  
  return (
    <div className="embla">
      <div className="embla__viewport" ref={emblaRef}>
        <div className="embla__container">
          {slides.map((item, index) => (
            <div className="embla__slide embla__class-names" key={index}>
              <div
                className="w-full h-full"
                style={{
                  backgroundImage: `url(${item.src})`,
                  backgroundPosition: "center",
                  backgroundSize: "cover",
                }}
              ></div>
            </div>
          ))}
        </div>
        <div className="embla__controls">
          <div className="embla__buttons">
            <div>
              <Button
                className="h-16 w-16 rounded-full bg-transparent text-white border-white"
                variant={"outline"}
                onClick={onPrevButtonClick}
                disabled={prevBtnDisabled}
              >
                <ChevronLeft className="h-7 w-7" />
                <span className="sr-only">Previous slide</span>
              </Button>
            </div>
            <div className="relative p-0 m-0">
              <CircularProgressbar 
                className="absolute w-[63px] -top-1 left-0 -z-10" 
                value={percentage} 
                styles={buildStyles({
                  pathColor: '#84bd00',
                  trailColor: '#ffffff00',
                })}
              />;
              <Button
                className="h-16 w-16 rounded-full bg-transparent text-white border-white"
                variant={"outline"}
                onClick={onNextButtonClick}
                disabled={nextBtnDisabled}
              >
                <ChevronRight className="h-7 w-7" />
                <span className="sr-only">Next slide</span>
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default EmblaCarousel;

```

**Comment by davidjerleke** [comment]
Hi @dpaulos6,

Thanks for your question. I can't debug fragments of code. Please create a CodeSandbox that reproduces the problem if you want me to look into it.

PS. About the urgency: I won't do anything faster or prioritise it higher because someone says it's urgent. I'm maintaining this project on my spare time so it's not exactly paying any of my bills. Please be respectful and don't ask me to hurry. This means no guarantees about when I can look into it.

Best,
David

**Reply by dpaulos6** [reply]
I didn't mean it like that, I would never ask you to prioritize me over someone else.

there's not an issue, I just don't know how to trigger the slide to the next item and how to listen to next/prev button clicks.
if you can provide two usage examples for these cases I would probably be able to work out the rest.

**Reply by davidjerleke** [reply]
@dpaulos6 no worries then.

[This example](https://www.embla-carousel.com/examples/predefined/#default) in the docs has a previous and next button implementation and it also disables it when it has reached the last slide. Click `Edit code` and open the file with the previous and next buttons.

Best,
David

**Reply by davidjerleke** [reply]
@dpaulos6 did the example I shared help you achieve what you want?
